lib/class-hook-reflector.php

```diff
@@ -11,40 +11,73 @@
 class Hook_Reflector extends BaseReflector {
 
 	/**
-	 * @return string
-	 */
-	public function getName() {
-		$printer = new PHPParser_PrettyPrinter_Default;
-		return $this->cleanupName( $printer->prettyPrintExpr( $this->node->args[0]->value ) );
-	}
-
-	/**
-	 * @param string $name
+	 * Hook names are the first argument to actions and filters.
+	 *
+	 * These are expected to be string values or concatenations of strings and variables.
+	 *
+	 * Example:
 	 *
+	 *     from: apply_filters( 'option_' . $option_name, $option_value );
+	 *     name: option_{$option_name}
+	 *
+	 *     from: do_action( 'wp_insert_post', $post_ID, $post, true );
+	 *     name: wp_insert_post
+	 *
+	 *     from: do_action( "{$old_status}_to_{$new_status}", $post );
+	 *     name: {$old_status}_to_{$new_status}
+	 *
+	 *     from: do_action( $filter_name, $args );
+	 *     name: {$filter_name}
+	 *
+	 * @param ?\PhpParser\Node $node Use this instead of the current node.
 	 * @return string
 	 */
-	private function cleanupName( $name ) {
-		$matches = array();
-
-		// quotes on both ends of a string
-		if ( preg_match( '/^[\'"]([^\'"]*)[\'"]$/', $name, $matches ) ) {
-			return $matches[1];
-		}
+	public function getName( $node = null ) {
+		/**
+		 * The current Node being examined.
+		 *
+		 * @var \PhpParser\Node
+		 */
+		$node    = $node ?? $this->node->args[0]->value;
```

Comment:
https://github.com/WordPress/phpdoc-parser/blob/7fc2227d2d4fb73f9f0b6e233413f3f9f9840e80/composer.json#L23

Reply:
@jrfnl can you provide some context or clarification for your comment?

are you suggesting we update the minimum supported version in this parser to [what WordPress supports](https://wordpress.org/about/requirements/), to 7.4?

Reply:
I'm not suggesting anything. I'm only pointing out that as things are at the moment, this change cannot go in.

Either the minimum PHP version needs to change or the syntax/functionality used needs to be made compatible with the minimum supported PHP version.

Reply:
updated the minimum version required in d57e2ed

Reply:
I don't know where you are getting your information from, but WP Core still has a [minimum supported PHP version of PHP 7.0](https://make.wordpress.org/core/handbook/references/php-compatibility-and-wordpress-versions/)....

Reply:
As a side-note: I believe that whether or it is acceptable to change the minimum PHP version of this package should be a separate discussion and should not be mixed in with a proposal for a (functional) change.

Reply:
> I don't know where you are getting your information from

that's why I left the link in the original message 😄 
but thanks for correcting me: I was reviewing the recommended setup, not the minimum.

> As a side-note: I believe that whether or it is acceptable to change the minimum PHP version of this package should be a separate discussion and should not be mixed in with a proposal for a (functional) change.

sounds like a good suggestion, which is why I asked if you meant something when you called out the line. I've updated this to avoid the `??` in 80ded80a9d1b7a48b3ef5964019b5cb72df0e309. thanks!